### PR TITLE
Login

### PR DESCRIPTION
**Pull request by benmorrismayden**

### CODE REVIEW

File: classes/DbConnector.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class DbConnector
+{
+
+    private $conn;
+
+    public function __construct()
+    {
+
+        $servername = "192.168.20.56";
```
Comment (aimeeew) [inline classes/DbConnector.php:11]: Single quotes?

Comment (aimeeew) [reply classes/DbConnector.php:11]: (and four more below)

File: classes/User.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * if email and password match database then sets $loggedIn to TRUE
+     *
+     * @param STRING $email user email
+     * @param STRING $password user password
+     *
+     * @throws Exception
+     */
+    function login($email, $password) {
+
+        $sql = "SELECT * FROM `users` WHERE `email` = :email;";
```
Comment (aimeeew) [inline classes/User.php:128]: Double quotes: 31, 37, 40, 43, 44, 66, 82

File: test/classes/UserTest.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    public function testSuccessfulLogin() {
+
+        // Note, details used relevant to testLogOut
+        $mockPDO = $this->createMock('PDO');
+        $mockStatement = $this->createMock('PDOStatement');
+        $mockStatement->method('fetch')->willReturn([
+            'id' => '1',
+            'email' => 'example@gmail.com',
+            'password' => sha1("1" . 'aaa')
```
Comment (aimeeew) [inline test/classes/UserTest.php]: double quotes line 17 and 60

Comment (aimeeew) [reply test/classes/UserTest.php]: & 41

File: classes/DbConnector.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+class DbConnector
+{
+
+    private $conn;
+
+    public function __construct()
```
Comment (aimeeew) [inline classes/DbConnector.php:8]: Needs docblocking

File: classes/DbConnector.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        $password = "";
+
+        try {
+            $this->conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password);
+        } catch (Exception $e) {
+            throw new Exception('error connecting to database');
+        }
+    }
+
+    public function getDB(){
```
Comment (aimeeew) [inline classes/DbConnector.php:24]: needs docblocking

File: classes/User.php [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+        if($user["password"] != $encryptPass) {
+            throw new Exception("incorrect email and password combination");
+        } else {
+            $this->loggedIn = TRUE;
+            $this->id = $user['id'];
+        }
+    }
+
+    public function logOut(){
```
Comment (aimeeew) [inline classes/User.php]: needs docblocking

File: classes/User.php [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        $this->loggedIn = FALSE;
+    }
+
+    public function getLoggedIn(){
```
Comment (aimeeew) [inline classes/User.php]: needs docblocking

File: test/classes/UserTest.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
+<?php
+
+require '/Users/benmorris/Documents/Development/academy/parking-system/classes/User.php';
+require '/Users/benmorris/Documents/Development/academy/parking-system/vendor/autoload.php';
+use PHPUnit\Framework\TestCase;
+
+class UserTest extends TestCase {
+
+    public function testSuccessfulLogin() {
```
Comment (aimeeew) [inline test/classes/UserTest.php]: docblocks

File: test/classes/UserTest.php [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        $user = new User($mockPDO);
+        try {
+            $user->login('example@gmail.com','aaa');
+        } catch (Exception $e) {
+        } finally {
+            $this->assertTrue($user->getLoggedIn());
+        }
+    }
+
+    public function testNonExistantUserLogin() {
```
Comment (aimeeew) [inline test/classes/UserTest.php]: docblocks

File: test/classes/UserTest.php [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+        $user = new User($mockPDO);
+        try {
+            $user->login('example@gmail.com','aaa');
+        } catch (Exception $e) {
+            $this->assertEquals("user does not exist", $e->getMessage());
+        }
+    }
+
+    public function testIncorrectPasswordLogin() {
```
Comment (aimeeew) [inline test/classes/UserTest.php]: docblocks

File: test/classes/UserTest.php [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+        $user = new User($mockPDO);
+        try {
+            $user->login('example@gmail.com','aaa');
+        } catch (Exception $e) {
+            $this->assertEquals("incorrect email and password combination", $e->getMessage());
+        }
+    }
+
+    public function testLogOut() {
```
Comment (aimeeew) [inline test/classes/UserTest.php]: docblocks

**Review by aimeeew — CHANGES_REQUESTED** [review]

File: test/classes/UserTest.php [hunk 12]
```diff
@@ -0,0 +1,88 @@
+<?php
+
+require '/Users/benmorris/Documents/Development/academy/parking-system/classes/User.php';
```
Comment (ghost) [inline test/classes/UserTest.php]: Absolute filepath - absolutely no no

Comment (ghost) [reply test/classes/UserTest.php]: actually, if you're autoloading, why require the class?

File: login.html [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    <meta charset="UTF-8">
+    <title>Title</title>
+</head>
+<body>
+    <div>
+        <br>
+        <div class="col-md-6 col-md-offset-3 well" id="login-form-div">
+            <h4>Please Login</h4>
+            <hr>
+            <form method="post" action="emailpasstet.php" id="loginForm" class="form-horizontal">
```
Comment (ArcticFoxJ) [inline login.html]: id/class name loginForm inconsistent with using - instead of cammelCase

File: index.php [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    try {
+        $user->validateToken($_SESSION['userAuth'], $_SESSION['id']);
+        $loggedIn = TRUE;
+    } catch(Exception $e) {
+//        session_destroy();
+        //do something
+    }
+
+
+}else {
```
Comment (ghost) [inline index.php]: Dodge formatting

File: test/classes/UserTest.php [hunk 15]
```diff
@@ -0,0 +1,74 @@
+<?php
+
+require __DIR__ . '/../../classes/User.php';
```
Comment (ghost) [inline test/classes/UserTest.php]: This should work without explicitly requiring the class(as that's the autoloaders job)

**Pull request merged**